### PR TITLE
refactor: improve build dts

### LINKED ISSUE #6338: [Bug Report] [TypeScript] [All] es/* should not use /// <reference types="node" />
Bug Type: **`TypeScript`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.0.3`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Safari/605.1.15`
- Build Tool: `Vite`

### Related Component

- `All`

### Reproduction Link

[Github Repo](https://github.com/exherb/vue-tsc)

### Steps to reproduce

pnpm install
cd dashboard
npm run lint

### What is Expected?

no lint error

### What is actually happening?

> @adtalos/dashboard@0.0.0 lint:vuetsc
> vue-tsc --noEmit

src/components/dynamic_material/download.ts:39:37 - error TS2551: Property 'pipeTo' does not exist on type 'ReadableStream'. Did you mean 'pipe'?

39   await new Blob([buffer]).stream().pipeTo(writable);
                                       ~~~~~~

  ../node_modules/.pnpm/@types+node@17.0.21/node_modules/@types/node/globals.d.ts:197:9
    197         pipe<T extends WritableStream>(destination: T, options?: { end?: boolean | undefined; }): T;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'pipe' is declared here.

Found 1 error.

### Additional comments

https://github.com/johnsoncodehk/volar/issues/989

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

closes #6338

add `type-safe` list to ensure that TS never reports an error.

### CODE REVIEW

File: build/types-definitions.ts [hunk 1]
```diff
@@ -1,19 +1,26 @@
...
 
 const TSCONFIG_PATH = path.resolve(projRoot, 'tsconfig.json')
 const outDir = path.resolve(buildOutput, 'types')
 
+const typeSafePaths = typeSafe.map((_path) => {
+  let safePath = path.resolve(projRoot, _path)
+  if (_path.endsWith('/')) safePath += path.sep
+  return safePath
+})
+
```
Comment (JeremyWuuuuu) [inline build/types-definitions.ts:23]: Could you please add a documentation here elaborating what is this all about?

Comment (sxzz) [reply build/types-definitions.ts:23]: Hmmmm? Is it really necessary? Currently the build system is maintained by me alone.

Comment (sxzz) [reply build/types-definitions.ts:23]: And this is only an alternative until the TS error is fully resolved.

Comment (JeremyWuuuuu) [reply build/types-definitions.ts:23]: No offense, that is why we need to document it.

Comment (sxzz) [reply build/types-definitions.ts:23]: Done.

**Review by jw-foss — APPROVED** [review]

**Pull request merged**